src/services/stats/stats-router.ts

```diff
@@ -0,0 +1,161 @@
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the number of priority attendees (staff only)
+statsRouter.get(
+    "/priority-attendee",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const currentTime = new Date();
+            const attendees = await Database.ATTENDEES.find({
+                priority_expiry: { $gt: currentTime },
+            });
+
+            return res.status(StatusCodes.OK).json({ count: attendees.length });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the attendance of the past n events (staff only)
+statsRouter.get(
+    "/attendance/:N",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const n = req.params.N;
+            if (!n) {
+                return res
+                    .status(StatusCodes.BAD_REQUEST)
+                    .json({ error: "MissingNParameter" });
+            }
+            const currentTime = new Date();
+            const events = await Database.EVENTS.find({
+                endTime: { $lt: currentTime },
+            })
+                .sort({ endTime: -1 })
+                .limit(parseInt(n));
+
+            const attendanceCounts = events.map(
+                (event) => event.attendanceCount
+            );
+
+            return res
+                .status(StatusCodes.OK)
+                .json({ attendanceCounts: attendanceCounts });
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+// Get the dietary restriction breakdown/counts (staff only)
+statsRouter.get(
+    "/dietary-restrictions",
+    RoleChecker([Role.enum.STAFF], false),
+    async (req, res, next) => {
+        try {
+            const attendees = await Database.ATTENDEES.find({});
+            let none = 0;
+            let dietary_restrictions = 0; // Gluten-Free, Lactose-Intolerant, No Pork, No Beef, No Fish, Halal, Vegetarian, Vegan, Diabetes
+            let allergies = 0; // Milk, Eggs, Tree nuts, Peanuts, Shellfish, Fish, Soy, Wheat, Sesame
+            let both = 0;
+            const dietary_restriction_counts: { [key: string]: number } = {};
+            const allergy_counts: { [key: string]: number } = {};
+
+            attendees.forEach((attendee) => {
+                if (
+                    attendee.dietary_restrictions.length > 0 &&
+                    attendee.allergies.length > 0
+                ) {
+                    both++;
+                } else if (attendee.dietary_restrictions.length > 0) {
+                    dietary_restrictions++;
+                } else if (attendee.allergies.length > 0) {
+                    allergies++;
+                } else {
+                    none++;
+                }
+
+                const attendee_dietary_restrictions: string[] =
+                    attendee.dietary_restrictions;
```

Comment:
no need to worry about both/none imo, just counts of most popular ones

Reply:
and then the counts of extra ones

Reply:
Im j gonna leave those there cuz it might be nice to have for the admin dashboard